### PR TITLE
Fix the datatable layout and component

### LINKED ISSUE #60: Fix the datatable layout and component
image

properly align the datatable component following is the reference image
image

### PR DESCRIPTION
**Pull request by aanish27**
fixed #60 @Raza9798 Please review

### CODE REVIEW

File: resources/views/admin.blade.php [hunk 1]
```diff
@@ -109,28 +103,48 @@
                 </div>
               </div>
 
-            <table id="myTable" class="table table-striped table-dark table-hover  " style="width: 100%"></table>
+            <table id="myTable" class="table table-striped table-dark table-hover " style="width: 100%" data-turbolinks="false"></table>
```
Comment (Raza9798) [inline resources/views/admin.blade.php:106]: explain why you have used `turbolinks` and make the row size smaller

image

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: resources/views/admin.blade.php (hunk 1 above)
Comment (aanish27) [reply resources/views/admin.blade.php:106]: i tried to edit the datatable layout styles by manipulating the bootstrap applied to it....that time i added this after referring to a post...cuz bootstrap wasn't new style weren't applying.....ig i forgot to remove it after i figured to changed the layout using data table..Ill remove it @Raza9798

**Review by Raza9798 — CHANGES_REQUESTED** [review]
image

* align the action buttons horizontally 
* there is a empty whitespace bar visible in the header section of the table
* text on the add button is not visible

please fix these changes

**Comment by aanish27** [comment]
> image
> 
> Datatable row should not be extended make it that as one line instead of multiple line of text

Made rows , single line data Please review @Raza9798  #60

**Review by Raza9798 — CHANGES_REQUESTED** [review]
image

making a common function to limit the character of the value if it is exceeded show the full value in tooltip or something. please remove the fixed width

**Comment by aanish27** [comment]
> image
> 
> making a common function to limit the character of the value if it is exceeded show the full value in tooltip or something. please remove the fixed width

Okay sure ill refactor the code

File: resources/views/admin.blade.php [hunk 2]
```diff
@@ -162,19 +165,52 @@
...
+                        { data: 'name' , title: 'Name' ,},
+                        { data: 'position' ,title:'Position'},
+                        { data: 'dob', title:'DOB' , width: '6%'},
+                        { data: 'email' , title:'Email'},
+                        { data: 'phone', title:'Phone' },
+                        { data: 'address' , title:'Address'},
+                        { data: 'bank_account.account_no', title:'Bank Acc' },
 
                     ],
+                    columnDefs: [
```
Comment (Raza9798) [inline resources/views/admin.blade.php:182]: Please do not use `columnDefs`. if we need to add a new column inside the table we have to change the target id of each columns you have setup.

**Review by Raza9798 — CHANGES_REQUESTED** [review]
requested changes

File: resources/views/admin.blade.php (hunk 2 above)
Comment (aanish27) [reply resources/views/admin.blade.php:182]: Yeah got it...ill implement in an alternate method...

File: resources/views/admin.blade.php [hunk 3]
```diff
@@ -225,9 +194,24 @@
                     });
                     table.draw(false);
                     $('.btn-close-dlt').click();
-                })
+                });
 
                 //hideLongText
+                function ellipsis(){
```
Comment (Raza9798) [inline resources/views/admin.blade.php:200]: remove addition functions like `ellipsis, truncateColumns`. using `hideLongText` method you can `substring` the values of  the data.

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: resources/views/admin.blade.php (hunk 3 above)
Comment (aanish27) [reply resources/views/admin.blade.php]: Okay sure removed please review again @Raza9798 #60

File: resources/views/admin.blade.php [hunk 4]
```diff
@@ -197,30 +201,18 @@
...
-                function ellipsis(){
-                    truncateColumns('column-company' , 26)
-                    truncateColumns('column-branch' , 15)
-                    truncateColumns('column-name' , 26)
-                    truncateColumns('column-position' , 20)
-                    truncateColumns('column-address' , 50)
-                }
-
-                function truncateColumns(columnName , charLimit){
+                function hideLongText(columnName , charLength){
```
Comment (Raza9798) [inline resources/views/admin.blade.php:204]: I noticed you call the `hideLongText` parent function where it has a child function which is looping over all the cells in specified columns.  which should be avoided, consider if we nested multiple tables. 

Please explain this method, and mention is there any alternatives available.

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: resources/views/admin.blade.php (hunk 4 above)
Comment (aanish27) [reply resources/views/admin.blade.php:204]: yes i can understand didn't give much thought at first...i tried so many options and methods..it seems liked nothing can be done without columnDef...so i implemented it using js...

hideLongText(columnName , charLength) is a simple function.....
it take all the rows of a column then it compares the text length f the content in the row with a given number. If its more than  that value it substring and creates a tool tip or else only tool tip...changes are done to the Dom element it self 

While i was searching i cam across the a best method to implement this feature using a extension built for data table but couldn't find a way to implement it without the target id...if its possible give me a hint please

https://datatables.net/plug-ins/dataRender/ellipsis


@Raza9798 #60

Comment (aanish27) [reply resources/views/admin.blade.php]: refactored the by referencing the example please review again #60  @Raza9798

**Review by Raza9798 — CHANGES_REQUESTED** [review]
the way you implemented is correct. but the purpose of overwriting a existing function is not necessary. please remove the overwrites and setup the default.

**Comment by aanish27** [comment]
> the way you implemented is correct. but the purpose of overwriting a existing function is not necessary. please remove the overwrites and setup the default.

sorry i cant understand , can you be more specific? @Raza9798

**Comment by Raza9798** [comment]
> > the way you implemented is correct. but the purpose of overwriting a existing function is not necessary. please remove the overwrites and setup the default.
> 
> sorry i cant understand , can you be more specific? @Raza9798

you have overwrite the existing method of datatable. remove the overwritten method

**Comment by aanish27** [comment]
> > > the way you implemented is correct. but the purpose of overwriting a existing function is not necessary. please remove the overwrites and setup the default.
> > 
> > 
> > sorry i cant understand , can you be more specific? @Raza9798
> 
> you have overwrite the existing method of datatable. remove the overwritten method

Yes got it...removed @Raza9798 please review again #60

File: resources/views/admin.blade.php [hunk 5]
```diff
@@ -144,12 +166,65 @@
...
+                        {
+                            data: 'name' ,
+                            title: 'Name' ,
+                            className: 'column-name',
+                            render: DataTable.render.ellipsis( 26 )
+                        },
+                        {
+                            data: 'position' ,
+                            title:'Position' ,
+                            className: 'column-position',
```
Comment (Raza9798) [inline resources/views/admin.blade.php:195]: please remove the `className` attributes which are not necessary

**Review by Raza9798 — CHANGES_REQUESTED** [review]

File: resources/views/admin.blade.php (hunk 5 above)
Comment (aanish27) [reply resources/views/admin.blade.php]: removed please review again @Raza9798 #60

**Review by Raza9798 — APPROVED** [review]

**Pull request merged**